Foundation/NSCFString.swift

```diff
@@ -62,11 +62,20 @@ internal final class _NSCFConstantString : _NSCFString {
         let ptr = unsafeAddress(of: self) + sizeof(OpaquePointer.self) + sizeof(Int32.self) + sizeof(Int32.self) + sizeof(_CFInfo.self)
         return UnsafePointer<UnsafePointer<UInt8>>(ptr).pointee
     }
+
+#if arch(s390x)
+    internal var _length : UInt64 {
+        let offset = sizeof(OpaquePointer.self) + sizeof(Int32.self) + sizeof(Int32.self) + sizeof(_CFInfo.self) + sizeof(UnsafePointer<UInt8>.self)
+        let ptr = unsafeAddress(of: self) + offset
+        return UnsafePointer<UInt64>(ptr).pointee
+    }
+#else
     internal var _length : UInt32 {
         let offset = sizeof(OpaquePointer.self) + sizeof(Int32.self) + sizeof(Int32.self) + sizeof(_CFInfo.self) + sizeof(UnsafePointer<UInt8>.self)
         let ptr = unsafeAddress(of: self) + offset
         return UnsafePointer<UInt32>(ptr).pointee
```

Comment:
Can we more tightly scope this conditional code, so as not to duplicate the rest?